### ISSUE TITLE
Carton intercepts --help, --version, and probably others

### ISSUE THREAD

**Issue by rkleemann** [perl-carton/carton#151]
When run with carton, my script cannot do anything with the `--help` and `--version` command parameters, as carton intercepts them and prints out it's own help and version information.

The command I'm using:  `carton exec ./my-script.pl --help`

Carton should stop processing command line options once it discovers an `exec`, as everything after that point should be passed, unmolested, to the process.

**Comment by maxhq** [comment]
I just stumbled upon this. The main problem IMHO is that from a user's point of view the behaviour changes depending on the arguments (s)he give to the script:
`carton exec bin/myscript --yo`works but
`carton exec bin/myscript --help` doesn't.

I think there are two reasons these two arguments could just be dropped:
1. `carton exec --version`and `carton exec --help` are most likely not used in any *scripts* in the wild
2. You can always do `carton help exec` which feels natural for e.g. `git` users.

**Comment by miyagawa** [comment]
Well, for now just use `--`, that will always work.

**Comment by s1037989** [comment]
Yes, `--` works for me.  Curious tho why it's intercepting `-n`?  I went thru all the lower case letters and it intercepted `-h` and `-v` obviously, but why `-n`?
